index.js

```diff
@@ -292,37 +303,47 @@ let rcl = {
     if (node.spinning) {
       throw new Error('The node is already spinning.');
     }
-    node.spinOnce(this._context.handle(), timeout);
+    node.spinOnce(node.context.handle(), timeout);
   },
 
   /**
    * @param {Context} context - The context to be shutdown.
    * @return {undefined}
    */
-  shutdown(context) {
-    if (!this._initialized) {
-      throw new Error('The module rclnodejs has been shut.');
+  shutdown(context = Context.defaultContext()) {
+    if (this.isShutdown(context)) {
+      throw new Error('The module rclnodejs has been shutdown.');
+      return;
     }
 
-    this._nodes.forEach(node => {
+    // check for non-existant or deleted context
+    // if (context === Context.defaultContext()) {
+    //   console.log('default ctx', this._nodes.has(Context.defaultContext()));
+    // }
+    if (!this._nodes.has(context)) {console.log('xxx'); return;}
```

Comment:
Could we have more descriptive wording here as log?

Reply:
removed cruft.